pmd-ui/src/main/java/net/sourceforge/pmd/util/fxdesigner/XPathPanelController.java

```diff
@@ -140,18 +150,19 @@ public void initialize(URL location, ResourceBundle resources) {
 
         xpathExpressionArea.addEventHandler(KeyEvent.KEY_PRESSED, t -> {
             if ((t.isControlDown() && t.getCode().isWhitespaceKey()) || t.getCode() == KeyCode.BACK_SPACE) {
```

Comment:
Intercepting the back space key here is suboptimal. For instance, if I remove a chunk of text with ctrl+backspace, the results are not highlighted properly.